File: 例子/python3的例子/hash.py
# -*- coding: utf-8 -*-
from netcacrypto import Hash
import sys

def get_hash_algo(name):
	if name=="MD5":
		return Hash.MD5
	elif name=="SHA1":
		return Hash.SHA1
	elif name=="SHA224":
		return Hash.SHA224
	elif name=="SHA256":
		return Hash.SHA256
	elif name=="SHA384":
		return Hash.SHA384
	elif name=="SHA512":
		return Hash.SHA512
	elif name=="SM3":
		return Hash.SM3
	elif name=="SHA512/224":
		return Hash.SHA512_224
	elif name=="SHA512/256":
		return Hash.SHA512_256
	elif name=="SHA3-224":
		return Hash.SHA3_224
	elif name=="SHA3-256":
		return Hash.SHA3_256
	elif name=="SHA3-384":
		return Hash.SHA3_384
	elif name=="SHA3-512":
		return Hash.SHA3_512	
	else:
		return -1

arg_count=len(sys.argv)

if arg_count!=3:
	print("Usage: python hash.py sha1|sha256 file_name")
	sys.exit(-1)

algo=get_hash_algo(sys.argv[1].upper())
if algo==-1:
	print("bad hash algo")
	sys.exit(-1)

with Hash(algo) as hash_obj: 
	with open(sys.argv[2],"rb") as f:
		while True:
			data=f.read(64*1024)
			if len(data)==0:
				break
			hash_obj.update(data)
	hash_value=hash_obj.final()
	print(hash_value.hex())

File: 例子/lua的例子/Hash.lua
local CryptCore=require("CryptCore")

function ToANSIString(utf8String)
    return CryptCore.UTF8ToACP(utf8String)
end

function Usage(scriptName)
    print(string.format("lua %s hashAlgo inFileName1 [... inFileNamen]",scriptName))
    print(ToANSIString("参数"))
    print(ToANSIString("\thashAlgo：Hash算法，比如:SHA-1、SM3之类"))
    print(ToANSIString("\tinFileName1：第一个文件名，其内容要进行Hash运算"))
	print(ToANSIString("\tinFileNamen：第n个文件名，其内容要进行Hash运算"))
    print(ToANSIString("说明"))
    print(ToANSIString("\t对文件inFileName1到文件inFileNamen进行Hash运算，Hash算法使用hashAlgo"))
end

function getHashAlgo(hashAlgo)
    hashAlgo=hashAlgo:upper()
    if hashAlgo== "SHA-1" or hashAlgo=="SHA1" then
        return CryptCore.Const.SHA1
    elseif  hashAlgo== "SM3" then
        return CryptCore.Const.SM3
    elseif  hashAlgo== "MD5" then
        return CryptCore.Const.MD5
    elseif hashAlgo== "SHA-224" or hashAlgo=="SHA224" then
        return CryptCore.Const.SHA224
    elseif hashAlgo== "SHA-256" or hashAlgo=="SHA256" then
        return CryptCore.Const.SHA256
    elseif hashAlgo== "SHA-384" or hashAlgo=="SHA384" then
        return CryptCore.Const.SHA384
    elseif hashAlgo== "SHA-512" or hashAlgo=="SHA512" then
        return CryptCore.Const.SHA512
    elseif hashAlgo== "SHA-512/224" then
        return CryptCore.Const["SHA-512/224"]
    elseif hashAlgo== "SHA-512/256" then
        return CryptCore.Const["SHA-512/256"]
    elseif hashAlgo== "SHA3-224" then
        return CryptCore.Const["SHA3-224"]
    elseif hashAlgo== "SHA3-256" then
        return CryptCore.Const["SHA3-256"]
    elseif hashAlgo== "SHA3-384" then
        return CryptCore.Const["SHA3-384"]
    elseif hashAlgo== "SHA3-512" then
        return CryptCore.Const["SHA3-512"]
    else
        return nil;
    end
end


function hashUpdate(hashObj,fileObj)
	local data
	local BLOCK_LEN=8192
	while (true) do
 		data=fileObj:read(BLOCK_LEN)
		if not data then
			break
		end

		hashObj:Update(data)
	end
end

function NewHash(hashAlgo)
    return CryptCore.NewHash(hashAlgo)
end

local argCount=#arg
if argCount<2 then
    Usage(arg[0])
    os.exit(-1)
end

local hashAlgo=getHashAlgo(arg[1])
if not hashAlgo then
    print(ToANSIString("未知Hash算法"));
    os.exit(-1)
end


local hashValue

hashObj=assert(NewHash(hashAlgo))
for i=2,argCount do
	fileName=arg[i]
	fIn=io.open(fileName,"rb")
	if fIn==nil then
		hashObj:Free()
		print(ToANSIString(string.format("打开文件(%s)失败",CryptCore.ACPToUTF8(fileName))));
		os.exit(-1)
	end
	hashUpdate(hashObj,fIn)
	fIn:close()
end

hashValue=hashObj:Final()
hashObj:Free()

print(ToANSIString(CryptCore.HexEncode(hashValue)))
